### PR TITLE
update global structs

### PR DESCRIPTION
**Pull request by Thomyrock**
just a few modifications to improve the structs used
correct calls to Isaac in collision solver

### CODE REVIEW

File: doc/global.var.doc.md [hunk 1]
```diff
@@ -11,11 +11,12 @@ Content of the struct "isaac" :
 | hp | 1 byte | health point of Isaac |
 | dmg | 1 byte | damage dealt by Isaac |
 | upgrades | 2 bytes | upgrades earned by Isaac (flags, to be defined) |
-| range | 1 byte | range of Isaac's tears |
+| range | 1 byte | 6 bits for range of Isaac's tears, 1 flag that tells if a tear was shot during the previous image |
```
Comment (DorianXGH) [inline doc/global.var.doc.md:14]: precise position of reserved bit, and revert the position of the flag and rage info, (basically use higher to low notation)

Comment (g33kex) [reply doc/global.var.doc.md:14]: Yes, we should be consistent with bit numbering. Bit 0 is low bit, bit 7 is high bit.

Comment (JeanMarieMineau) [reply doc/global.var.doc.md:14]: btw, please, use the MSBs first for the flags. it easier to retrieve the range when it's coded in the LSBs.

Comment (JeanMarieMineau) [reply doc/global.var.doc.md:14]: btw, please, use the MSBs first for the flags. it easier to retrieve the range when it's coded in the LSBs.

**Pull request merged**